mm2src/coins/lightning/ln_events.rs

```diff
@@ -114,22 +118,41 @@ impl EventHandler for LightningEventHandler {
                 funding_satoshis,
                 push_msat,
                 channel_type: _,
-            } => {
-                info!(
-                    "Handling OpenChannelRequest from node: {} with funding value: {} and starting balance: {}",
-                    counterparty_node_id,
-                    funding_satoshis,
-                    push_msat,
-                );
-                if self.channel_manager.accept_inbound_channel(temporary_channel_id, 0).is_ok() {
-                    // Todo: once the rust-lightning PR for user_channel_id in accept_inbound_channel is released
-                    // use user_channel_id to get the funding tx here once the funding tx is available.
-                }
-            },
+            } => self.handle_open_channel_request(*temporary_channel_id, *counterparty_node_id, *funding_satoshis, *push_msat),
         }
     }
 }
 
+pub async fn init_events_abort_handlers(
+    platform: Arc<Platform>,
+    db: SqliteLightningDB,
+) -> EnableLightningResult<Arc<PaMutex<Vec<AbortOnDropHandle>>>> {
+    let abort_handlers = Arc::new(PaMutex::new(Vec::new()));
+    let closed_channels_without_closing_tx = db.get_closed_channels_with_no_closing_tx().await?;
```

Comment:
Most of these loops if not all will do only 1 iteration and end straight away, the only loop here is in this function  https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_platform.rs#L402
in case there is an error from electrums. If I did a single loop it will be endless, so I am not sure which is the better solution.

Reply:
So `get_closed_channels_with_no_closing_tx` result is not empty only when node is restarted after being offline for some time?

Reply:
> So get_closed_channels_with_no_closing_tx result is not empty only when node is restarted after being offline for some time?

No, this happens when the node or mm2 is closed while waiting to find the closing tx https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_platform.rs#L404-L414 or if the internet or electrums are down when the node or mm2 is closed leading to this loop not returning the result https://github.com/KomodoPlatform/atomicDEX-API/blob/577fee5243e1a2468adde1556cec4d1a160795d0/mm2src/coins/lightning/ln_platform.rs#L402 
I guess this can be avoided if the closing transaction can be included in the `ChannelClosed` event from LDK side somehow. I will look at the LDK code and see the feasibility of this, if it can be done I might open an issue and a PR there.

Reply:
Added this to the issue's checklist https://github.com/KomodoPlatform/atomicDEX-API/issues/1045#issue-971642359